src/main/java/lotto/domain/Rank.java

```diff
@@ -0,0 +1,56 @@
+package lotto.domain;
+
+import java.util.Arrays;
+
+public enum Rank {
+    FIRST(6, 2_000_000_000L),
+    SECOND(5, 30_000_000L),
+    THIRD(5, 1_500_000L),
+    FOURTH(4, 50_000L),
+    FIFTH(3, 5_000L),
+    NONE(0, 0L);
+
+    private final int matchedCount;
+    private final long prizeMoney;
+
+    Rank(int matchedCount, long prizeMoney) {
+        this.matchedCount = matchedCount;
+        this.prizeMoney = prizeMoney;
+    }
+
+    public long calculateTotalPrizeMoney(int count) {
+        return prizeMoney * count;
+    }
+
+    public static Rank from(MatchDetail matchDetail) {
+        return Rank.of(matchDetail.matchedCount(), matchDetail.matchedBonus());
+    }
+
+    public static Rank of(int matchedCount, boolean isBonusNumberMatched) {
+        if (matchedCount == THIRD.matchedCount && !isBonusNumberMatched) {
+            return THIRD;
+        }
+
+        return Arrays.stream(values())
+                .filter(rank -> rank.matchedCount == matchedCount)
+                .findFirst()
+                .orElse(NONE);
+    }
```

Comment:
```suggestion
    public static Rank of(int matchedCount, boolean isBonusNumberMatched) {
        return Arrays.stream(values())
                .filter(rank -> rank.matchedCount == matchedCount)
                .filter(rank -> rank.matchedCount != SECOND.matchedCount || isBonusNumberMatched)
                .findFirst()
                .orElse(NONE);
    }
```
이렇게 filter를 추가하면 처음에 THIRD를 따로 반환하는 코드를 없앨 수 있습니다. 또는, 우선 변환한 뒤 SECOND만 조건을 따져 다운그레이드, 또는 업그레이드를 하는 방법은 어떨까요?

Reply:
가지를 쳐내려가는 것처럼 `early return`하는 방식이 조금 더 가독성이 좋을 것이라고 생각해서 이렇게 작성했어요. 처음에는 하나의 `stream`에 모든 순위를 가를 수 있게 고민해보았는데요, 같은 `matchedCount`를 가지는 두 개의 순위가 존재하다 보니 미리 하나의 가능성을 빼 두었습니다. 의견 고맙습니다!